x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeoPoint.java

```diff
@@ -58,7 +64,7 @@ protected NodeInfo<? extends Expression> info() {
         return NodeInfo.create(this, ToGeoPoint::new, field());
     }
 
-    @ConvertEvaluator(extraName = "FromString", warnExceptions = { IllegalArgumentException.class })
+    @ConvertEvaluator(extraName = "FromString", warnExceptions = { RuntimeException.class })
```

Comment:
Is it not better to change the parser to throw IllegalArgumentException? I did that in my other PR (reading from source), but we could view this as a bug-fix for 8.12 and do it there.

Reply:
See https://github.com/elastic/elasticsearch/pull/102880/files#diff-273e58854c0ce3de04bf9c2b0c750d1db8549254cda977f766e46e5d3244a5b6R81

Reply:
Done in 35cbe5a19676fd560665769b262279d4e01e8cfa

Reply:
> Is it not better to change the parser to throw IllegalArgumentException?

It is, IMO too, but wasn't sure why the `RuntimeException` was used in the 1st place, so adapted around it.